### PR TITLE
Wrap ns1 provider from xenolf/lego

### PR DESCRIPTION
**Pull request by scoursen**

### CODE REVIEW

File: ns1/ns1.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/mholt/caddy/caddytls"
+	"github.com/xenolf/lego/acme"
+	"github.com/xenolf/lego/providers/dns/ns1"
+)
+
+func init() {
+	caddytls.RegisterDNSProvider("ns1", NewDNSProvider)
+}
+
+// NewDNSProvider returns a new MyProvider DNS challenge provider.
```
Comment (mholt) [inline ns1/ns1.go]: What is "MyProvider"?

**Review by mholt — CHANGES_REQUESTED** [review]
Can you confirm what the environment variables are? Is it just NS1_API_KEY?

**Comment by scoursen** [comment]
Yes it is just NS1_API_KEY.  It is tied to your account so no need to
provide other credentials

File: ns1/ns1.go (hunk 1 above)
Comment (scoursen) [reply ns1/ns1.go]: I copy pasted from the sample in the wiki (https://github.com/mholt/caddy/wiki/Writing-a-Plugin:-DNS-Provider) and forgot to change the comment after adapting the code.

**Review by mholt — APPROVED** [review]

**Pull request merged**